modules/cluster-management/src/main/java/org/apache/ignite/internal/cluster/management/rest/exception/handler/IgniteInternalExceptionHandler.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.cluster.management.rest.exception.handler;
+
+import io.micronaut.context.annotation.Requires;
+import io.micronaut.http.HttpRequest;
+import io.micronaut.http.HttpResponse;
+import io.micronaut.http.annotation.Produces;
+import io.micronaut.http.server.exceptions.ExceptionHandler;
+import jakarta.inject.Singleton;
+import org.apache.ignite.internal.rest.api.ErrorResult;
+import org.apache.ignite.lang.IgniteInternalException;
+
+/**
+ * Handles {@link IgniteInternalException} and represents it as a rest response.
+ */
+@Produces
+@Singleton
+@Requires(classes = {IgniteInternalException.class, ExceptionHandler.class})
+public class IgniteInternalExceptionHandler implements ExceptionHandler<IgniteInternalException, HttpResponse<ErrorResult>> {
+
+    @Override
+    public HttpResponse<ErrorResult> handle(HttpRequest request, IgniteInternalException exception) {
+        ErrorResult errorResult = new ErrorResult("INITIALIZATION_EXCEPTION", exception.getMessage());
+        return HttpResponse.badRequest().body(errorResult);
```

Comment:
Are you sure it should be a "bad request` errror?

Reply:
Not it is not obvious what exactly happend on IgniteInternalException. `IgniteInternalException` looks like 500 and I think it is better to mark it as an internal server error.

Reply:
In future we have to introduce specific exceptions for situations like: already initialized, initializing, etc.

Reply:
> IgniteInternalException looks like 500 and I think it is better to mark it as an internal server error.

Exactly